src/dodal/devices/i10/diagnostics.py

```diff
@@ -0,0 +1,307 @@
+        stage_read_suffix: str = "STA",
+        stage_write_suffix: str = "CON",
+        name: str = "",
+    ) -> None:
+        with self.add_children_as_readables(Format.HINTED_SIGNAL):
+            self.stage_drop_down_set = epics_signal_rw(
+                stage_write_enum,
+                read_pv=prefix + stage_write_suffix,
+            )
+            self.stage_drop_down_readback = epics_signal_r(
+                stage_read_enum,
+                read_pv=prefix + stage_read_suffix,
+            )
+        super().__init__(name=name)
+
+
+class I10AravisDriverIO(AravisDriverIO):
+    """This is the standard webcam that can be found in ophyd_async
+    with four extra centroid signal. There is also a change in data_type due to it being
+    an older/different model"""
+
+    def __init__(
+        self,
+        prefix: str,
+        cam_infix: str = "CAM:",
+        stat_infix: str = "STAT:",
+        name: str = "",
+    ) -> None:
+        super().__init__(prefix + cam_infix, name)
+
+        # data type correction for i10 model.
+        self.data_type = epics_signal_r(
+            I10WebCamIODataType, prefix + cam_infix + "DataType_RBV"
+        )
+        # centroid x-y position
+        self.centroid_x = epics_signal_r(float, prefix + stat_infix + "CentroidX_RBV")
+        self.centroid_x_sigma = epics_signal_r(
+            float, prefix + stat_infix + "CentroidX_RBV"
+        )
+        self.centroid_y = epics_signal_r(float, prefix + stat_infix + "CentroidX_RBV")
+        self.centroid_x_sigma = epics_signal_r(
+            float, prefix + stat_infix + "CentroidY_RBV"
+        )
+
+
+class I10CentroidDetector(StandardReadable, Triggerable):
+    """Detector to read out the centroid position,
+    this is base off the SingleTriggerDetector in ophyd_async with added
+    readable default"""
+
+    def __init__(
+        self,
+        prefix: str,
+        name: str = "",
+    ) -> None:
+        self.drv = I10AravisDriverIO(prefix=prefix)
+        self.add_readables(
+            [self.drv.array_counter, self.drv.centroid_x, self.drv.centroid_y],
+            Format.HINTED_UNCACHED_SIGNAL,
+        )
+
+        self.add_readables([self.drv.acquire_time], Format.CONFIG_SIGNAL)
+
+        super().__init__(name=name)
+
+    @AsyncStatus.wrap
+    async def stage(self) -> None:
+        await asyncio.gather(
+            self.drv.image_mode.set(ImageMode.SINGLE),
+            self.drv.wait_for_plugins.set(True),
+        )
+        await super().stage()
+
+    @AsyncStatus.wrap
+    async def trigger(self) -> None:
+        await self.drv.acquire.set(True)
+
+
+class ScreenCam(Device):
+    """Compound device of pneumatic stage(fluorescent screen) and webcam"""
+
+    def __init__(
+        self,
+        prefix: str,
+        stage_write_enum: type[StrictEnum] = InOutTable,
+        stage_read_enum: type[StrictEnum] = InOutReadBackTable,
+        stage_read_suffix: str = "STA",
+        stage_write_suffix: str = "CON",
```

Comment:
Should: I can't see anywhere in the code currently where we use different values for `stage_write_enum`, `stage_read_enum`, `stage_read_suffix` or `stage_write_suffix` than the defaults. I would just hardcode them in below until you have a case where they are different as it leads to lots of boilerplate